hnn_core/network_builder.py

```diff
@@ -352,32 +368,31 @@ def __exit__(self, cell_type, value, traceback):
         if _LAST_NETWORK is not None:
             _LAST_NETWORK._clear_neuron_objects()
 
-    # this happens on EACH node
-    # creates self._gid_list for THIS node
     def _gid_assign(self):
-
+        """Assign cell IDs to this node"""
         self.net._update_cells()  # updates net.n_cells
 
-        rank = _get_rank()
-        nhosts = _get_nhosts()
-
+        n_hosts = _get_nhosts()
         # round robin assignment of gids
-        for gid in range(rank, self.net.n_cells, nhosts):
+        for gid in range(self._rank, self.net.n_cells, n_hosts):
             # set the cell gid
-            _PC.set_gid2node(gid, rank)
             self._gid_list.append(gid)
 
         # loop over all drives, then all cell types, then all artificial cells
         # only assign a "source" artificial cell to this rank if its target
         # exists in _gid_list. "Global" drives get placed on different ranks
         for drive in self.net.external_drives.values():
-            for conn in drive['conn'].values():  # all cell types
-                for src_gid, target_gid in zip(conn['src_gids'],
-                                               conn['target_gids']):
-                    if (target_gid in self._gid_list and
-                            src_gid not in self._gid_list):
-                        _PC.set_gid2node(src_gid, rank)
-                        self._gid_list.append(src_gid)
+            if drive['cell_specific']:
```

Comment:
I suppose I thought that this sort of gid allocation occurred earlier in the drives API. What is being changed here?

Reply:
The gids still need to get assigned to individual nodes when `MPIBackend` distributes them. Note that it does so differently when the drive is cell-specific.